File: app/src/main/java/com/emikhalets/datesdb/data/DateItem.java
package com.emikhalets.datesdb.data;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(tableName = "dates_table")
public class DateItem implements Serializable {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String name;
    private String date;
    private String type;

    public DateItem(String name, String date, String type) {
        this.name = name;
        this.date = date;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: app/src/main/java/com/emikhalets/datesdb/data/DateItem.kt
package com.emikhalets.datesdb.data

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "dates_table")
class DateItem : Serializable {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    var name: String
    var date: Long
    var type: String

    constructor(name: String, date: Long, type: String) {
        this.name = name
        this.date = date
        this.type = type
    }

    @Ignore
    constructor(id: Int, name: String, date: Long, type: String) {
        this.id = id
        this.name = name
        this.date = date
        this.type = type
    }
}